Ensure switcher does not display during site deletion

katinthehatsite opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Closes DOTCOM-14326

## Proposed Changes

This PR adjusts the display of the env switcher so that it does not show as soon as the staging site deletion starts and the user is redirected back to the production site. 

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Currently the switcher displays even after the deletion process is started and after the user is redirected to production from deletion modal, they can still switch to staging and they see the error. 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Pull the changes from this branch or use the Calypso Live Link
* Ensure that you have an Atomic site and a staging site
* Navigate to `/v2/sites/{yourAtomicSite}`
* Use the env switcher on the top to switch to your staging site
* Navigate to `Settings` tab on a staging site
* Initiate the staging site deletion
* Observe that once the modal closes and you are redirected to production site, you should not see the switcher anymore but should see the production badge only

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Ensure switcher does not display during site deletion

Commit:
Refactor

katinthehatsite commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -129,13 +132,19 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
+		...isDeletingStagingSiteQuery( otherEnvironmentSiteId ?? 0 ),
+		enabled: !! otherEnvironmentSiteId && otherEnvironment === 'staging',
+	} );
+
```
While working on this code, I see some interesting things so I will add them here and we can decide what to do next:

* I see that we set `isStagingSiteDeleting` to `true` as soon as the request to the API for staging site deletion succeeds. I checked the backend code and it seems that the deletion is happening via async job because the revert is necessary first and the endpoint does not return any data for when that async job is actually finished. This means that we redirect to the production site while the staging site is technically not deleted yet. 
* The reason why the switcher was looking stuck before is that when we redirect from the modal, it checks if there is a staging site, but it still returns `true` because the async job has not completed yet. We are not actively polling for when the staging site does not exist anymore, so the value does not update and the switcher looks stuck unless you refresh the page and we get the new data;
* I think we should ultimately keep on using `isStagingSiteDeleting` query so that we can have consistent UI updates (for example, switcher, sync button etc.) and so that we can cover for the state when the async job is running but I also think we would need to add some polling to make sure that we can set `isStagingSiteDeleting` correctly back to `false` (this is something that we are not doing yet). Perhaps, I am missing something obvious though. 

@ivan-ottinger what do you think?

katinthehatsite replied:
Alternatively, we can keep the user inside the modal longer and set it to loading state while the staging site is being deleted but the polling will be needed for that as well. Here, the question is - should the user be able to close the modal and if yes, then we are back with the same problem. I find this solution not ideal. We would need to coordinate if this type of behaviour is something that is  consistent with production sites.

ivan-ottinger replied:
Thank you for the ping, Kat! 

> This means that we redirect to the production site while the staging site is technically not deleted yet.

Yes, that's right. Since we don't need to bother user with the staging site deletion process (at least I think it doesn't add any value to them as they are getting rid of that site), redirection to the production site as soon as the deletion request is made seems like the right solution to me.

> I think we should ultimately keep on using isStagingSiteDeleting query so that we can have consistent UI updates (for example, switcher, sync button etc.) and so that we can cover for the state when the async job is running but I also think we would need to add some polling to make sure that we can set isStagingSiteDeleting correctly back to false (this is something that we are not doing yet). Perhaps, I am missing something obvious though.

I think this is a good approach. 👍🏼 One additional thought coming to my mind is that we will likely need a similar check for displaying the `+ Add staging site` button. cc @epeicher

katinthehatsite replied:
I added some polling in https://github.com/Automattic/wp-calypso/pull/105467/commits/ce76e0178154a4bb951e3db7e8b85a5983441b95

@epeicher @ivan-ottinger feel free to give this a test. It should work in the following way:

* Try deleting a staging site
* Observe that you are immediately redirected to the production when the API request succeeds 
* Observe that you don't see the switcher
* Wait 2-3 minutes (you can also check network requests at the same time)
* Observe that we are polling for the staging site existance
* Confirm that eventually you see the switcher without needed to refresh
* Confirm that you see the `Add staging site` in the switcher

katinthehatsite replied:
Sidenote: we discussed possibly keeping the switcher and just showing the `Add staging site` button as disabled during the deletion process but I think it is something we can sort out in the follow up PR

epeicher replied:
I think we could migrate the placeholder we did as part of https://github.com/Automattic/wp-calypso/pull/104795, as the user can always press the back button and interact with the staging site. This is just a suggestion as pressing the Back button is something that the user can do always, even after the deletion is completed.

Regarding the switcher being displayed after the deletion, I would suggest to display the switcher at all times and maybe add a `Deleting staging site...` as a placeholder for the `Add staging site`. Initially, it seems to me confusing to see the badge, and then suddenly the switcher, because as a user, I don't know if the deletion has happened immediately. Alternatively, we can add some feedback for this so the user knows that something is happening in the background 🤔 


https://github.com/user-attachments/assets/0c81c2ca-8ead-47f2-ad68-0e2aef1345aa

katinthehatsite replied:
> Regarding the switcher being displayed after the deletion, I would suggest to display the switcher at all times and maybe add a Deleting staging site... as a placeholder for the Add staging site. Initially, it seems to me confusing to see the badge, and then suddenly the switcher, because as a user, I don't know if the deletion has happened immediately. Alternatively, we can add some feedback for this so the user knows that something is happening in the background 🤔

Yeah we discussed this with @ivan-ottinger and I believe he is going to get in touch with Marina on this for what the good behaviour would be there. In this PR, I think we can focus on handling the UI around the async job and in the follow up tasks, adjust the switcher. It should be fairly straightforward afterwards with one or two lines  once all the conditions are correctly in place.

epeicher replied:
> Yeah we discussed this with @ivan-ottinger and I believe he is going to get in touch with Marina on this for what the good behaviour would be there. In this PR, I think we can focus on handling the UI around the async job and in the follow up tasks, adjust the switcher. 

That sounds like a great approach to me 👍

ivan-ottinger replied:
Yes, I am in communication with Marina regarding the transitions between states and will update you in related task / PR, or will create dedicated tasks for specific changes. 🙂

katinthehatsite replied:
Sounds good, I am now cleaning up this PR and rebasing on the newest trunk because of https://github.com/Automattic/wp-calypso/pull/105478 . I will mark this for review whenever I am done with the changes.

katinthehatsite replied:
For this bit:

> I think we could migrate the placeholder we did as part of #104795, as the user can always press the back button and interact with the staging site. This is just a suggestion as pressing the Back button is something that the user can do always, even after the deletion is completed.

Not sure what are the plans for this in the new dashboard, I will let @ivan-ottinger check that one!

ivan-ottinger replied:
> Not sure what are the plans for this in the new dashboard, I will let @ivan-ottinger check that one!

This is something I discussed with Marina already. We can indeed implement it, just will need to ensure it matches the design system of the new Hosting Dashboard. I am planning to create a separate project task for it so it does not distract us as it is lower priority.

Commit:
Add polling for actual staging site existance

katinthehatsite commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -129,13 +142,41 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
+	}, [ isStagingSiteDeleting, stagingSiteExistsFromQuery, stagingSiteId, queryClient ] );
+
+	const stagingSiteExists =
+		stagingSiteExistsFromQuery !== undefined ? stagingSiteExistsFromQuery : hasStagingSite( site );
```
since the query runs only when the staging site is being deleted, I am using `hasStagingSite( site )` for all other cases as it was used before

ivan-ottinger commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -129,13 +142,41 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
+		enabled: !! stagingSiteId,
+	} );
+
+	const productionSiteId = site.is_wpcom_staging_site
+		? site.options?.wpcom_production_blog_id ?? 0
+		: site.ID;
```
Here we could consider getting the `productionSiteId` with the help of `getProductionSiteId` (from `/utils/site-staging-site`).

katinthehatsite replied:
Adjusted in https://github.com/Automattic/wp-calypso/pull/105467/commits/f74cd0f894529a7b83fe3f417558422a07d77502

ivan-ottinger commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -129,13 +142,41 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
 		enabled: !! otherEnvironmentSiteId,
 	} );
 
+	const stagingSiteId = site.is_wpcom_staging_site ? site.ID : otherEnvironmentSiteId;
```
Here we could consider using `getStagingSiteId` (from `/utils/site-staging-site`).

katinthehatsite replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/105467/commits/7d8f62e69fa864bdd93aa67d6fcd1cb7214ae894

ivan-ottinger reviewed (commented):
Nice! Thank you for the changes, Kat.

I have noticed that when the site is being deleted, the switcher is replaced by the badge, and then the dropdown gets back with the `+ Add staging site` button:

<img width="1556" height="812" alt="Markup on 2025-09-01 at 17:16:16" src="https://github.com/user-attachments/assets/0124fc7d-90c0-4866-a68a-8ac8b0cb8639" />

That looks good to me as when I clicked the button, it did create a new staging site (even though we will need to clean up a couple things there outside of this PR - like the site creation progress or ensuring the `+ Add staging site` button is replaced by the `Staging` button as soon as the staging site is created).

I have left a couple of suggestions / considerations below.

ivan-ottinger commented:
> I have noticed that when the site is being deleted, the switcher is replaced by the badge, and then the dropdown gets back with the + Add staging site button:

Ah, just noticed you actually mention it in [your comment](https://github.com/Automattic/wp-calypso/pull/105467#discussion_r2314132723). 😄 Sorry, I missed it on the first read.

Commit:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into fix/env-switcher-availability-during-staging-deletion

Commit:
Fix queries imports

Commit:
Import back query client

Commit:
Use gerproductionsiteid

katinthehatsite committed:
Mark yolo as a code owner of staging sites (#105495)

* Mark yolo as a code owner of staging sites

* Add one more file

---------

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

Commit:
Use get staging site id

Commit:
Add clarification

Commit:
fix typo

gabrielcaires committed:
Add notification pages skeleton (#105485)

ivan-ottinger commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -133,13 +151,40 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
 		enabled: !! otherEnvironmentSiteId,
 	} );
 
+	const stagingSiteId = getStagingSiteId( site ) || otherEnvironmentSiteId;
```
Do we actually need the `|| otherEnvironmentSiteId` part? 🤔 `getStagingSiteId` should already return correct staging site ID regardless if the passed `site` is staging or production.

katinthehatsite replied:
This has been cleaned up as well 👍

ivan-ottinger commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -133,13 +151,40 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
+
+	const productionSiteId = getProductionSiteId( site ) || 0;
+
+	//Staging site deletion process runs via async job. We need to keep on polling for the staging site deletion before we start displaying the button to add a staging site again
```
```suggestion
	// Staging site deletion process runs via async job. We need to keep on polling for the staging site deletion before we start displaying the button to add a staging site again
```

katinthehatsite replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/105467/commits/f8f8d2b7f0d67f1cbeb6122277a52bbd7459ac80

ivan-ottinger commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -133,13 +151,40 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
+						! isStagingSiteDeleting &&
+						( stagingSiteExists ||
+							( otherEnvironmentSite && canManageSite( otherEnvironmentSite ) ) ||
+							( otherEnvironment === 'staging' && ! stagingSiteExists ) );
```
I might be missing something, but why is this condition required? 🤔

katinthehatsite replied:
This has also been simplified during the refactor.

nightnei committed:
Update codeowners for Emails related code (#105502)

ivan-ottinger commented on client/dashboard/sites/site/environment-switcher.tsx:
```diff
@@ -51,15 +61,22 @@ const EnvironmentSwitcherDropdown = ( {
 	onClose: () => void;
 } ) => {
 	const productionSite = otherEnvironment === 'staging' ? currentSite : otherEnvironmentSite;
-	const stagingSite = otherEnvironment === 'staging' ? otherEnvironmentSite : currentSite;
+	let stagingSite;
+	if ( otherEnvironment === 'staging' ) {
+		stagingSite = stagingSiteExists ? otherEnvironmentSite : undefined;
+	} else {
+		stagingSite = currentSite;
+	}
```
The following is not the goal of this PR, but I think that some logic in this file could be simplified by getting the production and staging site objects through `useQuery` - instead of using `otherEnvironment` checks):

```
	const { data: productionSite } = useQuery( {
		...siteByIdQuery( productionSiteId ?? 0 ),
		enabled: !! productionSiteId,
	} );
```

(but we can tackle it as a part of a separate PR)

katinthehatsite replied:
Actually, I was debugging it a bit more and I am not confident we are waiting long enough for this to complete 🤔 I am wondering if the staging site finished reverting in that case. I am going to check a couple of things and get through your comments after

katinthehatsite replied:
I think we are okay here, I checked the conditions for the banner display during deletion in the old dashboard and I think we are using the same approach.  👍

katinthehatsite replied:
Adjusted in https://github.com/Automattic/wp-calypso/pull/105467/commits/ef16b03e297c4a2d4d534988c8b925fb4c4a73e6 how we get the production and staging objects.

ivan-ottinger reviewed (commented):
Thank you for the updates, Kat!

I have left a couple comments below. Looks like we will also need a rebase and there's this one TypeScript error:

<img width="3312" height="1116" alt="Markup on 2025-09-02 at 12:41:24" src="https://github.com/user-attachments/assets/73be4562-370f-426b-bd0b-6d9416a9d863" />

m1r0 committed:
Hosting Dashboard: Add confirmation when deleting domain glue records (#105398)

* Add confirmation when deleting glue records

* Refactor the modal to use Dataview's `RenderModal`

* Improve the modal copy

* Rework component props

* Use `deleteMutation.isPending` instead of internal state

Commit:
Ensure we clean up queries

Commit:
Refactor how we get production and staging objects

Commit:
Fix comment

zaguiini committed:
Domain Search Rewrite: Link to existing flows within Hosting Dashboard (#105508)

* Remove full-page overlay from hosting dashboard

* Link to existing flows

m1r0 committed:
Domains: Add policy notices (#105363)

* Show policy notices on the featured domain suggestions

* Show badges instead of a notice on the domain suggestions

* Add a missing comma

* Refactor the v1 domain suggestions to show policy notice badges

* Add types for the policy notices

zaguiini committed:
Domain Search Rewrite: Add /setup/domain flow and link to it within Hosting Dashboard (#105510)

* Add new domain flow

* Link to new flow

bogiii committed:
Domains: Name Servers: Refactor the name servers component to use the new endpoint (#105458)

* NameServers form: Get rid of manual isWpcomNameServers check

* Rename data fields

* Replace manual nameServers check with the info from the BE

* Remove unused areAllWpcomNameServers function

* Optimistically update name servers

* Change the way of the optimistic data update

* Resolve conflicts

sirbrillig committed:
Hosting Dashboard: Add payment methods (#105388)

* Add formatCreditCardExpiry to datetime utils

* Fix StoredPaymentMethodRazorpay type

* Fix type of payment_type

* Fix getPaymentMethodImageURL to work for all paypal types

* Add i18n-utils to allowed packages for localizeUrl

* Move payment-methods directory to billing-payment-methods

* Add initial DataViews

* Allow toggling backup setting

* Add warning when missing tax information

* Move adjustDataViewFieldsForWidth to own file

* Adjust fields by width

* Add placeholder remove payment method dialog

* Add placeholder "Add payment method" button

* Make a PaymentMethodDeleteDialog

* Allow deleting payment methods

* Add editing tax_location form

* Add city, organization, address fields to edit dialog

* Update type imports for api-core

* Allow editing more than field in PaymentMethodEditDialog

* Fix import of queryClient in queries

* Fix query imports

* Use spacing prop for VStack gap

epeicher reviewed (approved):
Thanks Kat! I have tested it, and I think the main functionality is there, this is working as described and we can iterate based on this, so it LGTM!


https://github.com/user-attachments/assets/5fa20213-a62e-479b-a18e-9aa64e0724c7

epeicher committed:
Sync: Increase left padding on Sync button (#105506)

adimoldovan committed:
e2e tests: Configure Team City builds for Playwright Test: part 3 (#105499)

zaguiini committed:
StepContainerV2: Check for setup/domain manually to avoid Start flow clash (#105514)

brunobasto committed:
Dashboard v2: User Profile Page skeleton (#105491)

* Add gravatar logo for the gravatar public profile form section

* Add an "Update avatar" button to the EditGravatar component

* Add the Gravatar profile section component

* Add the personal details component

* Refactor the profile page to use separate cards for the "Personal details" and "Public Gravatar profile" sections

* Allow updating user's first name, last name and email on the updateProfile mutation

* Follow standard directory structure used in other pages

* Remove unnecessary scss file

* Move Gravatar-related components under ./profile-gravatar

epeicher committed:
Environment switcher: Update Add Staging site button

epeicher committed:
Revert "Environment switcher: Update Add Staging site button"

This reverts commit d1b328077b805740ed7b34aab61a4dc0d95c7bc9.

epeicher commented:
Hey @katinthehatsite, I pushed [d1b3280](https://github.com/Automattic/wp-calypso/pull/105467/commits/d1b328077b805740ed7b34aab61a4dc0d95c7bc9) into this branch by mistake, I have reverted it as part of [c5d5d20](https://github.com/Automattic/wp-calypso/pull/105467/commits/c5d5d20755d6c24bc9c4777812f253d72b4ab41f) so it should not affect this PR. Please let me know if you find any issues.

zaguiini committed:
Domain Search Rewrite: Port unavailable notice messages (#105519)

* Handle generic errors

* Move domain-search queries to api-queries package

* Use api-queries queryClient

* Port unavaiable messages code

zaguiini committed:
Domain Search Rewrite: Fix Calypso cart (#105520)

leonardost committed:
Domains Hosting Dashboard: Create agnostic TLD filter component (#105386)

* (WIP) Create agnostic TLD filter component

* (WIP) Create agnostic TLD filter component

* Remove code duplication

* Move popover component into its own file

* Encapsulate filter dropdown into its own component

* Create Filter component

* Move state functions out of the UI components

* Extract TLD list item into its own component

* Show labels in the available TLD list

* Add TLDs without leading dot in the filter

* Reorder properties, add comemnts

* Fix type import

* Update module exports

* Add optional onSubmit property to Filter

* Remove label from TLD input

* Update CSS to mirror production

* Replace DropdownFilter with the new agnostic Filter

* Add `showTldFilter` property

* Do not export components

* Replace `Filter` component with its internal UI components

* Move `filter` object to `DOmainSearchContext` and move state to `Filter` compoennt

* Make `FilterPopoverLabel` a `span` instead of `div`

* Update CSS colors to not use Calypso variables

* Make TLD in filter list `button`s instead of `div`s

* Remove keyboard listener from TLD element

* Fix CSS border issues

* Update comment

* Add filter and setFilter to context

* Remove comment

* Remove redundant property `showTldFilter`

* Rename `exactSldMatchesOnly` property to `exactMatchesOnly` in the new filter compoennts

* Move methods down from `Filter` to `DomainSearchControlsFilterPopover`

* Move more methods inside `filter-popover` and rename `exactMatchesOnly` back to `exactSldMatchesOnly`

* Move filter count processing to `DomainSearchControlsFilterButton`

* Remove more CSS in favor of WPDS components

* Inline the `handleTldsChange` function

* Move filter counting logic outside `filter-button` component

* Move `FilterState` to the `Filter` component folder

* Replace `span` for `Text` in `FilterPopoverLabel`

* Fix weird tyling for list of TLDs in hosting dashboard

* Import `FilterState` from the correct place in `filter-popover`

* Add `__next40pxDefaultSize` to buttons

* Reorder properties in `FilterPopoverLabel`

* Adjust radius of TLD list container Card border

* Simplify `FilterPopoverTld` component and CSS by using `Text` instead of `button`

* Move CSS class

* Center buttons text

* Make popover size consistent between Hosting Dashboard and Calypso

* Fix radius of TLD list container border

* Move method in file

* Render popover in the same hierarchy as the parent component

* Fix inline popover rendering in Calypso

* Move `temporaryFilter` inside popover component

* Remove `filter` and `setFilter` from the legacy DomainSearch context since they are not being used

* Remove duplicate imports

zaguiini committed:
Domain Search Rewrite: Move exact reasons to the package (#105521)

p-jackson committed:
Sync menu on HD overview uses DropdownMenu instead of raw Dropdown (#105492)

p-jackson committed:
User profile menu uses DropdownMenu instead of Dropdown (#105500)

lancewillett committed:
Jetpack Manage: fix unclear language for Complete plan description (#105522)

p-jackson committed:
Remove tooltip from reader button (#105498)

taipeicoder committed:
Hosting Dashboard: Implement ButtonGroup component (#105496)

taipeicoder committed:
Implement eligibility status (#105465)

arthur791004 committed:
Notifications v2: Mark note as read when selecting it (#105494)

* Notifications v2: Mark note as read when selecting it

* Address feedback

p-jackson committed:
Tidy type import (#105524)

p-jackson committed:
Show a "Hosting Dashboard" toggle in the v1 account settings (#105463)

dzver committed:
Add plans toggle between Builder and Hosting plans (#105390)

* Add a basic toggle on top of Plans

* A few more configs

* Show the deemphasize option and hide the Free plan

It's unstyled but there

* Styling

* Move the toggle above the stepper header

* Match Figma

* Align the copy to match Figma

* Hide pills in visual split

* Update the default intent

* Add right margin on mobile

* Update header and subheader according to the interim copy

* Add the new hosting features to the plans grid

* Add titles

* Update three features based on figma

* Sync the hosting features with figma

* Update the commerce features

Still unordered but reached my time limit

* Show free per request by Isa

* hide the second subheader

* Add ... or start with free to the new copy

* Fix the Free plan button to be inline

* Adjust pixels and colors to the latest changes

* Fine-tune borders, spacings

* Remove features per latest design request

* Undo changes that leaked to control

* Add experiment code

* Introduce the third variation

* Show toggle

* Term selection toggle showed up after a rebase

* Update features per latest request

Local dev environmnent no longer in use
Bring back 2 features

* Add tracks event

* Toggle simplification

* Nuria's preferred loading is with no selection

* Adjust styles per Nuria's request

* Fix a bug that Claudiu found in the plans grid

Clicking the intent tab on smaller screens messed up the comparison grid

Claudiu also proposed the solution

* Sentence casing requested by Tess

* Remove one use of the variation that is not needed

Props Claudiu

* The Figma has no free link in the hosting tab

* Add a way to trigger the modal from outside of PlansFeaturesMain

* Unbreak the modal

* Apply styles to the modal

* Remove rebase artifact

* use clsx per request by Christos

* Missed styles for the hosting intent

* Use clsx per request

* Remove styles from step container per Christos' request

* Rename class

* Fix regression spotted by Christos

The fix was created by Claudiu

* Rename feature per request

* Reduce dependebility on styles

* No inline casting

* Mobile first style

* Restore 1rem code

I accidentally removed it and it doesn't do anything so nothing changed
and I didn't notice

* Format the % per request

* Move file to lib/plans instead of lib/domains

* There's no longer a need to modify this file

* Remove 17px negative margin, no longer does anything useful

breaks the features list on mobile

* The title Storage is hidden in the design

ivan-ottinger reviewed (approved):
The changes look good and work as expected! 👌🏼🙂 

I can confirm the same experience as Roberto shared in his screen recording above.

I did not observe any regressions.

gius80 committed:
Hosting dashboard: Add domain transfer to other user (#105501)

* Create TransferDomainToAnyUser component

* Update layout

* Create delete transfer panel

* Fix query invalidation

* Rename functions

* Add site users query

* Work in progress

* Removed unused files

* Update component

* Fine tuning

* Finalize component

* Minor improvement

* Use memoization

* Fix transalations

Commit:
Add TODOs for when the flow for adding and deleting staging sites works correctly

Commit:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into fix/env-switcher-availability-during-staging-deletion
